### ISSUE TITLE
[MPS] `true_divide` backward seg fault

### ISSUE THREAD

**Issue by qqaatw** [pytorch/pytorch#82541]
### 🐛 Describe the bug

```python
device = "mps"
a = torch.tensor([[]], device=device, requires_grad=True)
b = torch.tensor([[]], device=device, requires_grad=True)
y = torch.true_divide(a, b)
y.sum().backward()  # seg fault
```
Found in  #82526.

cc @kulinseth @albanD 

### Versions

```
PyTorch version: 1.13.0a0+gitdb7187b
Is debug build: True
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: macOS 12.4 (arm64)
GCC version: Could not collect
Clang version: 13.1.6 (clang-1316.0.21.2.5)
CMake version: version 3.23.2
Libc version: N/A

Python version: 3.9.13 | packaged by conda-forge | (main, May 27 2022, 17:00:33)  [Clang 13.0.1 ] (64-bit runtime)
Python platform: macOS-12.4-arm64-arm-64bit
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: False

Versions of relevant libraries:
[pip3] mypy==0.960
[pip3] mypy-extensions==0.4.3
[pip3] numpy==1.21.6
[pip3] torch==1.13.0a0+git4eae04b
[pip3] torchvision==0.13.0
[conda] numpy                     1.21.6                   pypi_0    pypi
[conda] torch                     1.13.0a0+git4eae04b           dev_0    <develop>
[conda] torchvision               0.13.0                   pypi_0    pypi
```

**Comment by albanD** [comment]
No high pri on the crash as this is for empty Tensors that are an "edge case".
btw this is most likely the same root cause as the `sub` similar failure (I don't have the link to the PR on hand).

**Comment by qqaatw** [comment]
> No high pri on the crash as this is for empty Tensors that are an "edge case".
> 
> btw this is most likely the same root cause as the `sub` similar failure (I don't have the link to the issue on hand).

Yeah, I think so for the root cause.

**Comment by qqaatw** [comment]
Closing as this was fixed in #82650
